pkg/client/client_v2.go

```diff
@@ -278,7 +289,12 @@ func NewV2(target string, token string, versionInfo VersionInfo, opts ...V2Clien
 		o(&options)
 	}
 
+	if options.agentInfo != nil {
+		versionInfo.Version = options.agentInfo.Version
+	}
```

Comment:
This should be removed. Remove `Version` from `versionInfo`, its no longer used so we should not have any clients set it.